### PR TITLE
Update code for Laravel 5.8 release.

### PR DESCRIPTION
**Pull request by Schnoop**
Adopted changes that has been made to Laravel in 5.8 release.

### CODE REVIEW

**Comment by nWidart** [comment]
Hello,

I just pushed the 5.8 update too at the same time :P 
https://github.com/nWidart/laravel-modules on master branch

**Comment by Schnoop** [comment]
:D Bad luck. I've also adopted that changes made to the string and array helper methods and therefore would like to create a new release 5.0.
Also there are is a depreciation in the $defer car in the Service Providers.

File: src/ModulesServiceProvider.php [hunk 1]
```diff
@@ -9,12 +9,6 @@
 
 abstract class ModulesServiceProvider extends ServiceProvider
 {
-    /**
-     * Indicates if loading of the provider is deferred.
-     *
-     * @var bool
-     */
-    protected $defer = false;
```
Comment (nWidart) [inline src/ModulesServiceProvider.php]: Was this removed in laravel?

Comment (Schnoop) [reply src/ModulesServiceProvider.php]: > Deferred Service Providers
Likelihood Of Impact: Medium

The defer boolean property on the service provider which is/was used to indicate if a provider is deferred has been deprecated. In order to mark the service provider as deferred it should implement the Illuminate\Contracts\Support\DeferrableProvider contract.

**Comment by nWidart** [comment]
Thanks for the depreciation changes, applied those on master branch 👍

File: src/ModulesServiceProvider.php (hunk 1 above)
Comment (nWidart) [reply src/ModulesServiceProvider.php]: Oh nice ok.

**Comment by Schnoop** [comment]
Why did you drop PHP 7.1 support? Because of this, i've to stick to Laravel 5.7 :(

**Comment by nWidart** [comment]
Because phpunit 8 doesn't support 7.1 anymore.

**Comment by Schnoop** [comment]
There's no need to use PHPUnit 8. Laravel itself sticks to 7.

**Comment by nWidart** [comment]
Yes but I would like to use up to date packages 😄

**Pull request closed without merging**